test/extended/etcd/helpers/helpers.go

```diff
@@ -597,6 +600,114 @@ func MachineNameToEtcdMemberName(ctx context.Context, kubeClient kubernetes.Inte
 	return "", fmt.Errorf("unable to find a node for the corresponding %q machine on the following machine's IPs: %v, checked: %v", machineName, machineIPListSet.List(), nodeNames)
 }
 
+// MasterNodes returns a list of master nodes
+func MasterNodes(ctx context.Context, kubeClient kubernetes.Interface) []*corev1.Node {
+	masterNodes, err := kubeClient.CoreV1().Nodes().List(ctx, metav1.ListOptions{LabelSelector: masterNodeRoleLabel})
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	var nodes []*corev1.Node
+	for i := range masterNodes.Items {
+		node := &masterNodes.Items[i]
+		nodes = append(nodes, node)
+	}
+	return nodes
+}
+
+// NodeNameToMachineName finds a machine name that corresponds to the given node name
+// first it looks up a machine that corresponds to the node by comparing the ProviderID field
+//
+// # In cases the ProviderID is empty it will try to find a machine that matches an internal IP address
+//
+// note:
+// it will exit and report an error in case the machine was not found
+func NodeNameToMachineName(ctx context.Context, kubeClient kubernetes.Interface, machineClient machinev1beta1client.MachineInterface, nodeName string) (string, error) {
+	node, err := kubeClient.CoreV1().Nodes().Get(ctx, nodeName, metav1.GetOptions{})
```

Comment:
not that it matters much in a CI cluster, but wouldn't that be destined for your `MasterNodes` helper above?